### PR TITLE
Complete de milliseconds feature

### PR DESCRIPTION
**Pull request by fgalan**
This PR continues the work in PR #3668 with the following additions:

* Unify getNumberFieldAsDouble() and getNumberField() into one function, using the version introduced in PR #3668 (more flexible than the other version and working, as shown by the test regression) but with the simpler name "getNumberField".
* Add new .test with filter cases with milliseconds
* Fixes in .test  using CRE_DATE and MOD_DATE variables
* Fixes all the .test to use the ".000Z" when applicable (even if we support ".00Z" at attribute creation/updating time, it's better to use the same ".000Z" obtained in responses, for homogeneity)
* Modify documentation

### CODE REVIEW

**Comment by fgalan** [comment]
Travis is not activated in PR based on prelanding branches. Thus, I provide the trace of ut and ft runs on my local system (we will see the travis pass when we merge feature/miliseconds_in_timestamps-prelanding into master, in a next PR).

Unit tests:

```
[==========] 864 tests from 128 test cases ran. (56547 ms total)
[  PASSED  ] 864 tests.

  YOU HAVE 16 DISABLED TESTS
```

Funtional tests:

```
Total test time: 3943.83 seconds

2 test cases OK in the second attempt:
  o  0117_convop_using_standard_ops/getEntityTypes.test
  o  0740_crash_with_json_object_for_attributes/json_object_format.test

WARNING: 5 test cases disabled:
  o  2: 0000_bad_requests/exit.test
  o  30: 0000_ipv6_support/ipv4_only.test
  o  31: 0000_ipv6_support/ipv6_only.test
  o  390: 0917_queryContext_behaves_differently/query_with_and_without_forwarding.test
  o  625: 1310_suspect_200OK/suspect_200OK.test
```

File: doc/manuals/admin/database_model.md [hunk 1]
```diff
@@ -57,21 +57,21 @@ Fields:
     -   **mdNames**: an array of strings. Its elements are the names of the
         metadata of the attribute. Here the "." to "="
         replacement is not done.
-    -   **creDate**: the timestamp (as integer number) corresponding to attribute creation
-        (as a consequence of append).
-    -   **modDate**: the timestamp (as integer number) corresponding to last
-        attribute update. It matches creDate if the attribute has not been
-        modified after creation.
+    -   **creDate**: the timestamp (as float number, meaning seconds with milliseconds)
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md:60]:
```suggestion
    -   **creDate**: the timestamp (as a floating point number, meaning seconds with milliseconds)
```

File: doc/manuals/admin/database_model.md [hunk 2]
```diff
@@ -57,21 +57,21 @@ Fields:
...
+        corresponding to attribute creation (as a consequence of append).
+    -   **modDate**: the timestamp (as float number, meaning seconds with milliseconds)
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md:62]:
```suggestion
    -   **modDate**: the timestamp (as a floating point number, meaning seconds with milliseconds)
```

File: doc/manuals/admin/database_model.md [hunk 3]
```diff
@@ -57,21 +57,21 @@ Fields:
...
     attributes of the entity (without IDs). In this case, the "." to "="
     replacement is not done.
--   **creDate**: the timestamp (as integer number) corresponding
-    to entity creation date (as a consequence of append).
--   **modDate**: the timestamp (as integer number) corresponding to last entity update. Note
-    that it uses to be the same that a modDate corresponding to at least
-    one of the attributes (not always: it will not be the same if the
-    last update was a DELETE operation). It matches creDate if the
-    entity has not been modified after creation.
+-   **creDate**: the timestamp (as float number, meaning seconds with milliseconds)
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md:68]:
```suggestion
-   **creDate**: the timestamp (as a floating point number, meaning seconds with milliseconds)
```

File: doc/manuals/admin/database_model.md [hunk 4]
```diff
@@ -57,21 +57,21 @@ Fields:
...
+    corresponding to entity creation date (as a consequence of append).
+-   **modDate**: the timestamp (as float number, meaning seconds with milliseconds)
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md:70]:
```suggestion
-   **modDate**: the timestamp (as a floating point number, meaning seconds with milliseconds)
```

File: doc/manuals/admin/database_model.md [hunk 5]
```diff
@@ -57,21 +57,21 @@ Fields:
...
+    corresponding to last entity update. Note that it uses to be the same that a
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md:71]:
```suggestion
    corresponding to last entity update. Note that it is usually the same as a
```

File: test/functionalTest/cases/0000_deprecated_checkings/get_entity_dates_with_options.test [hunk 6]
```diff
@@ -65,8 +65,9 @@ echo
 echo "02b. Get entity with dates keyValues"
 echo "===================================="
 orionCurl --url /v2/entities/E?options=dateCreated,dateModified,keyValues
-CRE_DATE=$(echo "$_response" | python -mjson.tool | grep dateCreated | awk -F '"' '{print $4}')
-MOD_DATE=$(echo "$_response" | python -mjson.tool | grep dateModified | awk -F '"' '{print $4}')
+# Note we remove milliseconds part, as it makes comparison with date output harder
```
Comment (kzangeli) [inline test/functionalTest/cases/0000_deprecated_checkings/get_entity_dates_with_options.test:68]:
```suggestion
# Note that we remove the milliseconds part, as this makes comparison with date output easier
```

File: test/functionalTest/cases/0000_deprecated_checkings/get_entity_dates_with_options.test [hunk 7]
```diff
@@ -120,8 +121,9 @@ echo
 echo "05b. Get entity with dates keyValues"
 echo "===================================="
 orionCurl --url /v2/entities/E?options=dateCreated,dateModified,keyValues
-CRE_DATE=$(echo "$_response" | python -mjson.tool | grep dateCreated | awk -F '"' '{print $4}')
-MOD_DATE=$(echo "$_response" | python -mjson.tool | grep dateModified | awk -F '"' '{print $4}')
+# Note we remove milliseconds part, as it makes comparison with date output harder
```
Comment (kzangeli) [inline test/functionalTest/cases/0000_deprecated_checkings/get_entity_dates_with_options.test:124]:
```suggestion
# Note that we remove the milliseconds part, as this makes comparison with date output easier
```

File: test/functionalTest/cases/0876_attribute_dates/attribute_dates.test [hunk 8]
```diff
@@ -55,17 +55,18 @@ echo
 
 echo "02. GET E1/A1, see dateCreated == dateModified"
 echo "=============================================="
+# Note we remove milliseconds part, as it makes comparison with date output harder
```
Comment (kzangeli) [inline test/functionalTest/cases/0876_attribute_dates/attribute_dates.test:58]:
```suggestion
# Note that we remove the milliseconds part, as this makes comparison with date output easier
```

File: test/functionalTest/cases/0876_attribute_dates/attribute_dates.test [hunk 9]
```diff
@@ -97,20 +98,21 @@ echo
 
 echo "04. GET E1/A1, check dateModified and dateCreated"
 echo "================================================="
+# Note we remove milliseconds part, as it makes comparison with date output harder
```
Comment (kzangeli) [inline test/functionalTest/cases/0876_attribute_dates/attribute_dates.test:101]:
```suggestion
# Note that we remove the milliseconds part, as this makes comparison with date output easier
```

File: test/functionalTest/cases/0876_entity_dates/get_entity_dates.test [hunk 10]
```diff
@@ -65,8 +65,9 @@ echo
 echo "02b. Get entity with dates keyValues"
 echo "===================================="
 orionCurl --url '/v2/entities/E?options=keyValues&attrs=dateCreated,dateModified,*'
-CRE_DATE=$(echo "$_response" | python -mjson.tool | grep dateCreated | awk -F '"' '{print $4}')
-MOD_DATE=$(echo "$_response" | python -mjson.tool | grep dateModified | awk -F '"' '{print $4}')
+# Note we remove milliseconds part, as it makes comparison with date output harder
```
Comment (kzangeli) [inline test/functionalTest/cases/0876_entity_dates/get_entity_dates.test:68]:
```suggestion
# Note that we remove the milliseconds part, as this makes comparison with date output easier
```

File: test/functionalTest/cases/0876_entity_dates/get_entity_dates.test [hunk 11]
```diff
@@ -120,8 +121,9 @@ echo
 echo "05b. Get entity with dates keyValues"
 echo "===================================="
 orionCurl --url '/v2/entities/E?options=keyValues&attrs=dateCreated,dateModified,*'
-CRE_DATE=$(echo "$_response" | python -mjson.tool | grep dateCreated | awk -F '"' '{print $4}')
-MOD_DATE=$(echo "$_response" | python -mjson.tool | grep dateModified | awk -F '"' '{print $4}')
+# Note we remove milliseconds part, as it makes comparison with date output harder
```
Comment (kzangeli) [inline test/functionalTest/cases/0876_entity_dates/get_entity_dates.test:124]:
```suggestion
# Note that we remove the milliseconds part, as this makes comparison with date output easier
```

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**